code/dictation.py

```diff
@@ -23,33 +38,45 @@ def word(m) -> str:
 @mod.capture(rule="({user.vocabulary} | <phrase>)+")
 def text(m) -> str:
     """A sequence of words, including user-defined vocabulary."""
-    return format_phrase(m)
+    return apply_formatting(m)
 
-@mod.capture(rule="({user.vocabulary} | {user.punctuation} | <phrase>)+")
+@mod.capture(rule="({user.vocabulary} | {user.punctuation} | {user.prose_snippets} | <phrase> | {user.prose_modifiers})+")
 def prose(m) -> str:
     """Mixed words and punctuation, auto-spaced & capitalized."""
-    text, _state = auto_capitalize(format_phrase(m))
-    return text
+    return apply_formatting(m).replace("“", "\"").replace("”", "\"")
+
+@mod.capture(rule="({user.vocabulary} | {user.punctuation} | {user.prose_snippets} | <phrase>)+")
+def raw_prose(m) -> str:
+    """Mixed words and punctuation, auto-spaced & capitalized, without quote straightening and commands (for use in dictation mode)."""
+    return apply_formatting(m)
 
 
 # ---------- FORMATTING ---------- #
-def format_phrase(m):
-    words = capture_to_words(m)
+def apply_formatting(m):
+    formatter = DictationFormat()
+    formatter.state = None
     result = ""
-    for i, word in enumerate(words):
-        if i > 0 and needs_space_between(words[i-1], word):
-            result += " "
-        result += word
-    return result
-
-def capture_to_words(m):
-    words = []
     for item in m:
-        words.extend(
-            actions.dictate.replace_words(actions.dictate.parse_words(item))
-            if isinstance(item, grammar.vm.Phrase) else
-            item.split(" "))
-    return words
+        words = None
+        if item == "cap":
+            formatter.force_cap = True
+            formatter.force_no_cap = False
+        elif item == "no caps":
+            formatter.force_no_cap = True
+            formatter.force_cap = False
+        elif item == "no space":
+            # Typically helpful to reset capitalization as well.
+            formatter.reset()
+            formatter.force_no_space = True
+        elif isinstance(item, grammar.vm.Phrase):
+            words = actions.dictate.replace_words(actions.dictate.parse_words(item))
+        else:
+            words = item.split(" ") if item != " " else [" "]
```

Comment:
This is orthogonal to this PR, but I think we could get away with setting `words = [item]` here. The auto-capitalization code doesn't care about begin handed words one-by-one, so splitting words out is not really necessary. This would automatically preserve whitespace instead of needing to have a special-case for it.

Reply:
Agreed -- I kept this as is since I was already changing so much and I didn't notice any problems with the current implementation. Happy to make this change here if you would like.